snmp/datadog_checks/snmp/config.py

```diff
@@ -184,20 +192,37 @@ def add_profile_tag(self, profile_name):
         self.tags.append('snmp_profile:{}'.format(profile_name))
 
     @staticmethod
-    def create_snmp_engine(mibs_path=None):
-        # type: (str) -> Tuple[SnmpEngine, MibViewController]
+    def _create_mib_builder(mibs_path):
+        # type: (Optional[str]) -> Tuple[MibBuilder, MibInstrumController, MibViewController]
+        mib_builder = MibBuilder()
+        if mibs_path:
+            mib_builder.addMibSources(DirMibSource(mibs_path))
+        mib_instrum = MibInstrumController(mib_builder)
+        mib_view = MibViewController(mib_builder)
+        return mib_builder, mib_instrum, mib_view
+
+    @classmethod
+    def _get_mib_builder(cls, mibs_path, shared_mib_builder):
+        # type: (Optional[str], bool) -> Tuple[MibBuilder, MibInstrumController, MibViewController]
+        if not shared_mib_builder:
+            return cls._create_mib_builder(mibs_path)
+        # Use a lock to make sure we don't concurrently create the cached objects
+        with cls._mib_lock:
+            if cls._mib_builders.get(mibs_path) is None:
+                cls._mib_builders[mibs_path] = cls._create_mib_builder(mibs_path)
+            return cls._mib_builders[mibs_path]
```

Comment:
Can we _maybe_, _perhaps_ move this to a tightly scoped `MIBLoader` component of some sort?

I think isolating hacks into components that can be understood in isolation has been a nice pattern for us in terms of keeping the "happy path" code readable.

Reply:
@therve Put a quick sub-PR together here https://github.com/DataDog/integrations-core/pull/6734 to attempt to tackle this :+1: